MicroPython/src/cloud/AWS_cloud.py

```diff
@@ -23,24 +22,23 @@ def device_configuration(self, data: list[dict]) -> int:
         :return: Error code (0 - OK, 1 - Error).
         """
 
-        config.cfg.access_points = data
-        config.cfg.save()
-
         logging.info("Wifi access point configuration:")
 
         for access_point in data:
             logging.info("Ssid: {} Password: {}".format(access_point["ssid"], access_point["password"]))
 
         wireless_controller = wirerless_connection_controller.get_wireless_connection_controller_instance()
         try:
-            utils.connect_to_wifi(wireless_controller)
+            utils.connect_to_wifi(wireless_controller, data)
             logging.info(wireless_controller.sta_handler.ifconfig())
             self.configure_aws_thing()
+            config.cfg.access_points = data
+            config.cfg.save()
```

Comment:
Saving config is not needed here as if exception occurs config will be saved with default access_points and if it won't it will be saved in line '44' after setting ap_config_done to True